ee/tabby-ui/app/(dashboard)/settings/team/components/user-table.tsx

```diff
@@ -243,3 +223,127 @@ export default function UsersTable() {
+  const onOpenChange = (open: boolean) => {
+    if (submitting) return
+    setOpen(open)
+  }
+  const { copyToClipboard, isCopied } = useCopyToClipboard({
+    timeout: 1000
+  })
+  const generateResetPasswordUrl = useMutation(generateResetPasswordUrlMutation)
+  const handleGenerateResetPassworkURL: MouseEventHandler<
+    HTMLButtonElement
+  > = e => {
+    e.preventDefault()
+    if (submitting) return
+
+    setSubmitting(true)
+    generateResetPasswordUrl({ userId: user.node.id })
+      .then(res => {
+        const link = res?.data?.generateResetPasswordUrl
+        if (link) {
+          copyToClipboard(link)
+          setOpen(false)
+        } else {
+          toast.error(
+            res?.error?.message || 'Failed to generate password reset link'
+          )
+        }
+      })
+      .catch(error => {
+        toast.error(error?.message || 'Failed to generate password reset link')
+      })
+      .finally(() => {
+        setSubmitting(false)
+      })
+  }
+
+  useEffect(() => {
+    if (isCopied) {
+      toast.success('Password reset link copied to clipboard')
+    }
+  }, [isCopied])
+
+  return (
+    <>
+      <DropdownMenu modal={false}>
+        <DropdownMenuTrigger asChild>
+          <Button size="icon" variant="ghost">
+            <IconMore />
+          </Button>
+        </DropdownMenuTrigger>
+        <DropdownMenuContent collisionPadding={{ right: 16 }}>
+          {!!user.node.active && (
+            <DropdownMenuItem
+              onSelect={() => onUpdateUserRole(user.node)}
+              className="cursor-pointer"
+            >
+              <span className="ml-2">
+                {user.node.isAdmin ? 'Downgrade to member' : 'Upgrade to admin'}
+              </span>
+            </DropdownMenuItem>
+          )}
+          {!!user.node.active && (
+            <DropdownMenuItem
+              onSelect={() => onUpdateUserActive(user.node, false)}
+              className="cursor-pointer"
+            >
+              <span className="ml-2">Deactivate</span>
+            </DropdownMenuItem>
+          )}
+          {!user.node.active && (
+            <DropdownMenuItem
+              onSelect={() => onUpdateUserActive(user.node, true)}
+              className="cursor-pointer"
+            >
+              <span className="ml-2">Activate</span>
+            </DropdownMenuItem>
+          )}
+          <DropdownMenuItem
+            onSelect={() => setOpen(true)}
+            className="cursor-pointer gap-1"
+          >
+            <span className="ml-2">Initate resetting password</span>
+          </DropdownMenuItem>
+        </DropdownMenuContent>
+      </DropdownMenu>
+      <AlertDialog open={open} onOpenChange={onOpenChange}>
+        <AlertDialogContent>
+          <AlertDialogHeader>
+            <AlertDialogTitle>Initiate resetting password</AlertDialogTitle>
+            <AlertDialogDescription>
+              Are you sure you want to create a password reset link for{' '}
```

Comment:
By clicking "Yes," a password reset link will be generated for <span>...</span>. The password won't be modified until the user follows the instructions in the link to make the change.

Reply:
<img width="569" alt="image" src="https://github.com/user-attachments/assets/ec427ec7-24c5-45a7-ab47-f72221cc3fca">